context/context.go

```diff
@@ -85,7 +85,7 @@ func (r *ResolvedRemotes) BaseRepo(io *iostreams.IOStreams) (ghrepo.Interface, e
 		return nil, err
 	}
 
-	if len(repos) == 0 {
+	if len(repos) <= 1 && len(r.remotes) > 0 {
 		return r.remotes[0], nil
 	}
```

Comment:
Maybe this is an obscure edge case but what about a case where there are multiple remotes and they only resolve to a single repo? Returning `r.remotes[0]` seems like it could result in the wrong remote getting selected.

```
if len(repos) == 1 {
    return repos[0], nil
}

// I would perhaps even return an error here saying no remotes resolved to GitHub repos?
if len(repos) == 0 {
    return r.remotes[0]
}
```

Reply:
I'm not worried about it. The shape of the RepoNetwork is directly related to the list of remotes locally. Multiple remotes all resolving to a single repo is a strange enough edge-case that I don't think we have to handle it.

Reply:
That one does seem unlikely. What about:

Remote 1 -> does not resolve to any GitHub repos because of auth issue or it is malformed remote or the repo was archived/deleted/renamed/etc
Remote 2 -> resolves to one GitHub repo

This is the one I was more concerned about. Normally you would think that an error would pop up and we wouldn't reach this point but the `RepoNetwork` query tolerates repo not found errors and so a remote can result in not resolving to any repo at all.

Going to approve the PR as the code looks good. Feel free to address this edge case if you feel it is relevant.

Reply:
ah, I see, thanks for elaborating; I wasn't interpreting your edge case correctly.

I do think there is merit in handling this after all so I'll push a commit for that.